### DISCUSSION TITLE
SV06 Ace

### DISCUSSION THREAD

**Discussion by scotthep** [bassamanator/Sovol-SV06-firmware#198] [Show and tell]
New owner of an SV06 Ace and a Linux fanboy.  Sovol moved to OrcaSlicer which is a mess on linux.  In ArchLinux, the AUR build fails and the AppImage fails to run.  Same with Debian 12.  Although I have some basic coding knowledge, I'm new to g-code/Aurdino.
I was able to build the latest stable version, v2.90, in Debian 12.  Debian packages PrusaSlicer, but v2.50 does not have Klipper support.

I used an online *.json -> *.ini converter and, after deleting the duplicate lines, imported it and got a snippet of the start g-code.

The snipet was the last two lines of the following code.  Looking at the *json file for 0.4 nozzles/PLA I saw additional start g-code that I copy/pasted into the PrusaSlicer profile editor.

`M140 S[bed_temperature_initial_layer_single] ;set bed temp
M190 S[bed_temperature_initial_layer_single] ;wait for bed temp
G90;
G1 X0 F6000;
G1 Y20;
G1 Z0.400 F600;
G1 Y-4 F6000;
M400;
M104 S[nozzle_temperature_initial_layer] ;set extruder temp
M109 S[nozzle_temperature_initial_layer]; wait for extruder temp
G91;
M83;
G1 E-0.200 Z5 F600\nG1 X{print_bed_max[1] / 3} F6000;
G1 Z-4.800 F600;
G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.24} F3000;
G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.16} F3000;
G1 Y1 E0.16 F3000;
G1 X-{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.16} F3000;
G1 X-{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.24} F3000;
G1 Y1 E0.24 F3000\nG1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.24} F3000;
G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.16} F3000;
G1 E-0.200 Z1 F600;
M400;
G28 ; home all axes
G1 Z5 F5000 ; lift nozzle`

Hopefully this will be a good starting point to develop an SV06Ace profile.  I have some time to put into this but need some hand-holding/guidance.

Some additional code needed:
1.  Runout sensor - found a post in Discussions and will add.
2. Nozzle wipe.  The SV06Ace has a silicon brush at the back of the plate.  The code is in the provided SV06Ace benchy.  I suspect the existing code that wipes on the plate will work.
3. The SV06Ace has one big part cooling fan.  I think the older SV06 has an additional fan in the tool head.

Constructive criticism appreciated.

[start_gcode.txt](https://github.com/user-attachments/files/18267323/start_gcode.txt)

**Comment by bassamanator** [comment]
I'm on EndeavourOS, so Arch, and the latest OrcaSlicer appimage runs fine (though I'm not an OS user).

Can you try running the appimage from the command line and paste the output?

For those looking, the Klipper config, and all the available macros can be found [here](https://github.com/Sovol3d/SV06-ACE/blob/main/sovol/printer_data/config/Macro.cfg).

**Comment by bassamanator** [comment]
Here's the exact `start gcode` pulled from official Sovol OS profile. Most of this should simply have been included in the `START_PRINT` macro in order to reduce the chance for user error, but here we are
```
START_PRINT
M140 S[bed_temperature_initial_layer_single] ;set bed temp
M190 S[bed_temperature_initial_layer_single] ;wait for bed temp
G90
G1 X0 F6000
G1 Y20
G1 Z0.400 F600
G1 Y-4 F6000
M400
M104 S[nozzle_temperature_initial_layer] ;set extruder temp
M109 S[nozzle_temperature_initial_layer];wait for extruder temp
G91
M83
G1 E-0.200 Z5 F600
G1 X{print_bed_max[1] / 3} F6000
G1 Z-4.800 F600
G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.24} F3000
G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.16} F3000
G1 Y1 E0.16 F3000
G1 X-{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.16} F3000
G1 X-{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.24} F3000
G1 Y1 E0.24 F3000
G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.24} F3000
G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.16} F3000
G1 E-0.200 Z1 F600
M400
```

`end gcode` is simply `END_PRINT`.

If you post the `START_PRINT` macro (`[gcode_macro START_PRINT]`) here, should be easy to extract the nozzle wipe.

**Comment by scotthep** [comment]
Thanks.

Upstream orcaslicer fixed a compilation error on Arch Linux:
https://github.com/SoftFever/OrcaSlicer/commit/5115ceb8f876e9921e6d8d24666e39f3e8a831d1

I modified the AUR orca-slicer-git PKGBUILd to pull the latest code base
and it built and runs.  The SV06 Ace profile was in the code:
https://aur.archlinux.org/packages/orca-slicer-git

I can send you the Arch binary if you're interested.

Gonna try to build in Debian 12 later today.  It is a long build.

Thanks for providing alternative profiles for PrusaSlicer.

On Dec 28, 2024: 20:04, Bassam wrote:
>I'm on EndeavourOS, so Arch, and the latest OrcaSlicer appimage runs fine
>(though I'm not an OS user).
>
>Can you try running the appimage from the command line and paste the output?
>
>—
>Reply to this email directly, view it on GitHub, or unsubscribe.
>You are receiving this because you authored the thread.*Message ID: 
>***@***.***
>.com>
>

-- 
J. Scott Heppler

Penguin Innovations

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 


NOTICE: This e-mail message and any attachments may
contain legally privileged and confidential information intended
solely for the use of the intended recipients. If you are not an
intended recipient, you are hereby notified that you have
received this message in error and any review, dissemination,
distribution, copying, or other unauthorized use of this email
and any attachment is strictly prohibited. If you have received
this email in error, please notify the sender immediately and
delete the message and any attachments from your system.

**Comment by scotthep** [comment]
Pulled from the 3DBench g-code:

2 Thumbnail *.png's blocks generated by OrcaSlicer:
  ; thumbnail begin 96x96 4740

  ; thumbnail begin 160x160 930

The resolution differs from PrusaSlicer.

There appears to be some camera code:
M106 S0
;LAYER_CHANGE
;Z:0.25
;HEIGHT:0.25
TIMELAPSE_TAKE_FRAME

Wipe g-code:
;WIPE_START
G1 X84.728 Y103.457 E-.06308
G1 X84.93 Y103.336 E-.0589
G1 X85.312 Y103.187 E-.10254
G1 X85.751 Y103.113 E-.11132
G1 X86.407 Y103.073 E-.16417
;WIPE_END


On Dec 28, 2024: 21:07, Bassam wrote:
>Here's the exact start gcode pulled from official Sovol OS profile. Most of
>this should simply have been include in the START_PRINT macro, but here we are
>
>START_PRINT
>M140 S[bed_temperature_initial_layer_single] ;set bed temp
>M190 S[bed_temperature_initial_layer_single] ;wait for bed temp
>G90
>G1 X0 F6000
>G1 Y20
>G1 Z0.400 F600
>G1 Y-4 F6000
>M400
>M104 S[nozzle_temperature_initial_layer] ;set extruder temp
>M109 S[nozzle_temperature_initial_layer];wait for extruder temp
>G91
>M83
>G1 E-0.200 Z5 F600
>G1 X{print_bed_max[1] / 3} F6000
>G1 Z-4.800 F600
>G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.24} F3000
>G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.16} F3000
>G1 Y1 E0.16 F3000
>G1 X-{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.16} F3000
>G1 X-{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.24} F3000
>G1 Y1 E0.24 F3000
>G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.24} F3000
>G1 X{print_bed_max[1] / 6} E{print_bed_max[1] / 6 * 0.16} F3000
>G1 E-0.200 Z1 F600
>M400
>
>end gcode is simply END_PRINT.
>
>If you post the START_PRINT macro ([gcode_macro START_PRINT]) here, should be
>easy to extract the nozzle wipe.
>
>—
>Reply to this email directly, view it on GitHub, or unsubscribe.
>You are receiving this because you authored the thread.*Message ID: 
>***@***.***
>.com>
>

-- 
J. Scott Heppler

Penguin Innovations

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - 


NOTICE: This e-mail message and any attachments may
contain legally privileged and confidential information intended
solely for the use of the intended recipients. If you are not an
intended recipient, you are hereby notified that you have
received this message in error and any review, dissemination,
distribution, copying, or other unauthorized use of this email
and any attachment is strictly prohibited. If you have received
this email in error, please notify the sender immediately and
delete the message and any attachments from your system.

**Comment by liberodark** [comment]
Hi, @scotthep 

You can install Orca Slicer with flatpak : 

```
flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
wget https://github.com/SoftFever/OrcaSlicer/releases/download/nightly-builds/OrcaSlicer-Linux-flatpak_V2.3.0-dev.flatpak
flatpak install ./OrcaSlicer-Linux-flatpak_V2.3.0-dev.flatpak
flatpak run io.github.softfever.OrcaSlicer
```

or with nix : 

`nix-shell -p orca-slicer --run orca-slicer`

Best Regards

**Comment by scotthep** [comment]
I modified the AUR orca-slicer-git PKGBUILd to pull the latest code base
and it built and runs fine.

**Comment by scotthep** [comment]
With the new Debian 13 release, I tried to build the latest OrcaSlicer nighly release and it has many, many issues.  It was segfaulting on the initial run for 2 weeks, yesterday the latest finally ran albeit with many warnings about font width.  The issues on OrcaSlicers github page are increasing rapidly, ~ 12% over the last 3 weeks.

I redouble my efforts for a PrusaSlicer profile.  The earlier code in this thread did not start without homing the head first.  If the END_CODE entry is left blank, the nozzle will remain positioned over the last layer.

I've attached my SVO6_ACE_.ini file as a text file.  Rename the text file  to SV06_ACE.ini to import into prusaslicer.  Note that my settings are not mainstream - I'm printing a scalpel handle with high resolution and solid infill.    Once imported you may want to revert to default print settings.  Or you can copy/paste the gcode from the prusaslicer_sv06ace_gcode_works.txt file into the custom G-Code Printer setting.

Edit:  Updated GCode File
[PS_SV06_Ace_GCODE.txt](https://github.com/user-attachments/files/24983355/PS_SV06_Ace_GCODE.txt)

**Comment by scotthep** [comment]
I saw the addition of the SV06 Ace to your SV06 Firmware and the start/stop codes for PrusaSlicer - thanks.
I did have one question on the Start and End Codes.  In PrusaSlicer, are they sufficient in and of themselves or do I need to add some of the macros, and any other code, to my stock Klipper configuration?
